app/scripts/controllers/edit/route.js

```diff
@@ -75,26 +75,55 @@ angular.module('openshiftConsole')
             };
 
             DataService.list("services", context).then(function(resp) {
+              $scope.loading = false;
+
               var servicesByName = resp.by("metadata.name");
               var to = _.get(route, 'spec.to', {});
-              $scope.loading = false;
-              $scope.services = orderByDisplayName(servicesByName);
+
+              // Make sure there is an option in the route editor for a service
+              // even if it doesn't exist. The editor will warn if the service
+              // is not found.
+              var ensureService = function(objectReference) {
+                if (servicesByName[objectReference.name]) {
+                  return;
+                }
+
+                // Make a copy to avoid mutating the DataService cache.
+                servicesByName = angular.copy(servicesByName);
+
+                // Add a dummy service since the `osc-routing` directive
+                // expects an object, not just a name. TODO: Update the
+                // osc-routing directive to take in a name rather than an
+                // object.
```

Comment:
@jwforres I might see how hard it is to change the directive rather than making a dummy object since I think that's cleaner, although it's a larger code change.